Bio/Phenotype/PhenMicro.py

```diff
@@ -0,0 +1,1101 @@
+        WellRecord('(0.0, 10.0), (0.25, 14.0), (0.5, 19.0), (0.75, 19.0), (1.0, 23.0)...(71.75, 32.0)')
+
+        Note that long time spans are shown truncated.
+        """
+        lines = []
+        if self.plate and self.plate.id:
+            lines.append("Plate ID: %s" % self.plate.id)
+        if self.id:
+            lines.append("Well ID: %s" % self.id)
+        lines.append("Time points: %i" % len(self))
+        lines.append("Minum signal %.2f at time %.2f" % 
+                    min(self, key=lambda x: x[1]))
+        lines.append("Maximum signal %.2f at time %.2f" % 
+                    max(self, key=lambda x: x[1]))
+        lines.append(repr(self))
+        return "\n".join(lines)
+        
+    def get_raw(self):
+        """Get a list of time/signal pairs"""
+        return [(t,self._signals[t]) for t in sorted(self._signals.keys())]
+        
+    def get_times(self):
+        """Get a list of the recorded time points"""
+        return sorted(self._signals.keys())
+        
+    def get_signals(self):
+        """Get a list of the recorded signals (ordered by collection time)"""
+        return [self._signals[t] for t in sorted(self._signals.keys())]
+        
+    def fit(self, function=None):
+        """Fit a sigmoid function to this well and extract curve parameters
+        
+        If function is not set, the following order will be applied:
+        * gompertz
+        * logistic
+        * richards
+        
+        The first function that is succesfuly fitted to the signals will
+        be used to extract the curve parameters; if no function can be fitted
+        a warning is raised.
+        
+        Some of the parameters don't need the curve fitting, which depends on
+        the scipy curve_fit function. If scipy is not available, only those
+        paraeters will be calculated, and a warning will be raised."""
+        avail_func = ('gompertz', 'logistic', 'richards', )
+        
+        if function not in avail_func and function is not None:
+            raise ValueError('Sigmoid function %s not recognized'%function)
+            
+        # Parameters not dependent on curve fitting
+        self.max = max(self, key=lambda x: x[1])[1]
+        self.min = min(self, key=lambda x: x[1])[1]
+        
+        self.average_height = np.array( self.get_signals() ).mean()
+        
+        # Parameters that depend on scipy curve_fit
+        try:
+            from pm_fitting import fit, get_area
+        except ImportError:
+            warnings.warn('SciPy not installed, could not calculate area, '+
+                          'plateau, slope, lag, v and y0',
+                           ImportWarning)
+            return
+        
+        self.area = get_area(self.get_signals(), self.get_times())
+        
+        if function is None:
+            sigmoid = avail_func
+        else:
+            sigmoid = (function, )
+        
+        mod = __import__ ('Bio.Phenotype.pm_fitting')
```

Comment:
This worries me - in part given subtle Python 2/3 changes in imports. Would a dictionary of functions work better?